app/api/annotator.py

```diff
@@ -102,7 +102,8 @@ def post(self):
         image_model.update(
             set__metadata=image.get('metadata', {}),
             set__annotated=annotated,
-            set__category_ids=image.get('category_ids', [])
+            set__category_ids=image.get('category_ids', []),
+            set__regenerate_thumbnail=annotated
```

Comment:
Every time the user saves in the annotator, the flag is set to regenerate the thumbnail. Unless you want to compare each annotation to check if anything has changed, this is probably the best option...